src/server/parser/tokeniser.ts

```diff
@@ -453,21 +453,25 @@ export class Tokeniser
 				result = '\x07'
 				break
 			case 'u':
-			case 'U':
+			case 'U': {
+				// `readHexToken` sets token type to `hexLit` so we need to change it back to original type
+				const type = this._token.type
 				this.readHexToken()
+				this._token.set(type, this._token.value)
 				return String.fromCodePoint(Number.parseInt(this._token.value, 16))
-			default:
-				// TODO: return Result
-				console.warn(`Unknown unicode character: ${this.currentChar}`)
```

Comment:
It might be a good idea for now to keep this warning to help with extension debugging and eventually to turn into some kind of error squiggly to help the programmer. This seems fine otherwise